operator/src/main/java/oracle/kubernetes/operator/helpers/KubernetesUtils.java

```diff
@@ -141,4 +102,33 @@ static V1ObjectMeta getResourceMetadata(Object resource) {
       return new V1ObjectMeta();
     }
   }
+
+  /**
+   * Returns true if the first metadata indicates a newer resource than does the second. 'Newer'
+   * indicates that the creation time is later. If two items have the same creation time, a higher
+   * resource version indicates the newer resource.
+   *
+   * @param first the first item to compare
+   * @param second the second item to compare
+   * @return true if the first object is newer than the second object
+   */
+  public static boolean isFirstNewer(V1ObjectMeta first, V1ObjectMeta second) {
+    if (second == null) return true;
+    if (first == null) return false;
+
+    DateTime time1 = first.getCreationTimestamp();
+    DateTime time2 = second.getCreationTimestamp();
+
+    if (time1.isAfter(time2)) {
+      return true;
+    } else if (time2.isAfter(time1)) {
+      return false;
+    } else {
+      return getResourceVersion(first) > getResourceVersion(second);
+    }
```

Comment:
The original version you had for isNewer compares version numbers even if the timestamp of time2 'isAfter' time1 or time2 equals time1, and returns true if if the second version number is higher.   The new code only compares version numbers if time1 equals time2.   Is this intentional?

In other words, here's the logic of the new code (simplified a little vs. what you have above):

if (time1.isEquals(time2)) {
  return getResourceVersion(first) > getResourceVersion(second);
} else {
  return time1.isAfter(time2);
}

And here's the logic of the old code you had before:

if (time2.isAfter(time1)) {
  return false;
} else if (getResourceVersion(second) < getResourceVersion(first)) {
  return true;
} else {
  return time2.isBefore(time1);
}

Reply:
Yes, thank you. That is an improvement on what I had, and does exactly the same thing. Resource versions are just a tie-breaker for when the times are equal.